fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/tree/PullUpPredicateRule.java

```diff
@@ -0,0 +1,269 @@
+            return null;
+        }
+
+        @Override
+        public Void visitLogicalJoin(OptExpression optExpression, RewriteContext context) {
+            LogicalJoinOperator logicalJoinOperator = (LogicalJoinOperator) optExpression.getOp();
+
+            OptExpression leftChild = optExpression.inputAt(0);
+            RewriteContext leftChildRewriteContext = new RewriteContext();
+            leftChild.getOp().accept(this, leftChild, leftChildRewriteContext);
+            if (logicalJoinOperator.getJoinType().isRightOuterJoin()) {
+                for (BinaryPredicateOperator binaryOp : leftChildRewriteContext.operatorSet) {
+                    leftChild = OptExpression.create(new LogicalFilterOperator(binaryOp), leftChild);
+                    optExpression.setChild(0, leftChild);
+                }
+                leftChildRewriteContext.clear();
+            } else {
+                OptExpression c = handleLegacyPredicate(optExpression.getOutputColumns(), leftChild, leftChildRewriteContext);
+                optExpression.setChild(0, c);
+            }
+
+            OptExpression rightChild = optExpression.inputAt(1);
+            RewriteContext rightChildRewriteContext = new RewriteContext();
+            rightChild.getOp().accept(this, rightChild, rightChildRewriteContext);
+            if (logicalJoinOperator.getJoinType().isLeftOuterJoin()) {
+                for (BinaryPredicateOperator binaryOp : rightChildRewriteContext.operatorSet) {
+                    rightChild = OptExpression.create(new LogicalFilterOperator(binaryOp), rightChild);
+                    optExpression.setChild(1, rightChild);
+                }
+                rightChildRewriteContext.clear();
+            } else {
+                OptExpression c = handleLegacyPredicate(optExpression.getOutputColumns(), rightChild, rightChildRewriteContext);
+                optExpression.setChild(1, c);
+            }
+
+            for (BinaryPredicateOperator binaryPredicateOperator : leftChildRewriteContext.operatorSet) {
+                context.put(binaryPredicateOperator);
+            }
+
+            for (BinaryPredicateOperator binaryPredicateOperator : rightChildRewriteContext.operatorSet) {
+                context.put(binaryPredicateOperator);
+            }
+
+            return null;
+        }
+
+        // At present, the Filter under the Window/Repeat/Table Function node is no longer pulled upwards.
+        // If there is such a demand case in the future, it can be continuously optimized
+
+        @Override
+        public Void visitLogicalWindow(OptExpression optExpression, RewriteContext context) {
+            convertPredicateToLogicalFilter(optExpression, context);
+            return null;
+        }
+
+        @Override
+        public Void visitLogicalRepeat(OptExpression optExpression, RewriteContext context) {
+            convertPredicateToLogicalFilter(optExpression, context);
+            return null;
```

Comment:
maybe predicate can't pass through repeat node, there will cause column to nullable